SimpleNotePad.java

```diff
@@ -0,0 +1,225 @@
+import java.awt.Dimension;
+import java.awt.Graphics;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.awt.print.PageFormat;
+import java.awt.print.Printable;
+import java.awt.print.PrinterException;
+import java.awt.print.PrinterJob;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.stream.Stream;
+
+import javax.swing.JFileChooser;
+import javax.swing.JFrame;
+import javax.swing.JMenu;
+import javax.swing.JMenuBar;
+import javax.swing.JMenuItem;
+import javax.swing.JOptionPane;
+import javax.swing.JScrollPane;
+import javax.swing.JTextPane;
+
+/*
+ * Code Smell Refactoring:
+ * -Changed the variables to private instead of public
+ * -Changed variable names at the beginning of class to make it more readable
+ * -Line 145: Removed undo feature since it is not in the design specs.
+ * -Split the file and edit menu items to separate methods to make it more
+ * 	readable.
+ * -Removed the println in the paste block. 
+ * -Moved some of the code from the if-else block in the actionPerformed method
+ *  to separate methods to make it more readable.
+ * -Removed unused code from paste function
+ * 
+ * -Added new features in the assignment spec
+ * 
+ */
+
+public class SimpleNotePad extends JFrame implements ActionListener{
+	private JMenuBar menuBar;
+	private JMenu fileMenu, editMenu, openRecentFile;
+	private JTextPane d = new JTextPane();
+	private JMenuItem newFile, saveFile, printFile, openFile;
+	private JMenuItem copy, paste, simpleReplace;
+	private JMenuItem[] recent = new JMenuItem[5];

```

Comment:
```suggestion
	private final static JMenuItem[] recent = new JMenuItem[5];
```
The data is (probably) supposed to be shared between all instances of SimpleNotePad, so this field should be marked as static. The maximum number of files the array can store should not change during execution, so this field should also be marked as static.

(This is just something to think about for the future, as it is well beyond the scope of this assignment, but this feature would need to handle persistence of the array between sessions (i,e, file I/O) and propagating any changes the menu to all running instances of the application to be complete for real-world purposes).